Robot2019/src/main/java/frc/robot/lib/logging/LogFiles.java

```diff
@@ -0,0 +1,186 @@
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVPrinter;
+
+import edu.wpi.first.wpilibj.DriverStation;
+
+/**
+ * Creates the required log files and initializes logging code
+ */
+final class LogFiles {
+    
+    private static int logId;
+    private static ArrayList<Integer> existingLogIds;
+    private static String dirString = System.getProperty("user.home") + "/logs";
+    private static File dirFile, infoFile;
+
+    /**
+     * Initializes the logging code or returns if it is already initialized
+     * @param dataFormat The {@link CSVFormat} to use when logging data
+     */
+    static void init(CSVFormat dataFormat) {
+        if(GlobalLogInfo.isInit()) {
+            return;
+        }
+        dirFile = new File(dirString);
+        try {
+            dirFile.createNewFile();
+        } catch(IOException e) {
+            LogUtils.handleLoggingApiDisableError("creating log directory", e);
+            return;
+        }
+        infoFile = new File(dirString + "/info.txt");
+        try {
+            infoFile.createNewFile();
+        } catch(IOException e) {
+            LogUtils.handleLoggingApiDisableError("creating log directory", e);
+            return;
+        }
+        existingLogIds = new ArrayList<>();
+        try {
+            findExistingLogIds();
+            clearOldLogs();
+            findLogId();
+            createLogFiles(dataFormat);
+        } catch(AbortException e) {
+            LogUtils.handleLoggingApiDisableError(e.getMessage(), (Exception)e.getCause());
+            return;
+        }
+    }
+
+    private static void findExistingLogIds() {
+        existingLogIds.clear();
+        File[] files = dirFile.listFiles();
+        for(File file: files) {
+            String name = file.getName();
+            if(name.indexOf(".") == -1) {
+                continue;
+            }
+            String idString = file.getName().substring(0, name.indexOf("."));
+            try {
+                int id = Integer.parseInt(idString);
+                if(!existingLogIds.contains(id)) {
+                    existingLogIds.add(id);
+                }
+            } catch(NumberFormatException e) {
+                continue;
+            }
+        }
+    }
+
+    private static void clearOldLogs() throws AbortException {
+        int lines = 0;
+        ArrayList<String> linesD = new ArrayList<>();
+        try(BufferedReader br = new BufferedReader(new FileReader(infoFile))) {
+            String line;
+            while((line = br.readLine()) != null) {
+                linesD.add(line);
+                lines++;
+            }
+        } catch(IOException e) {
+            throw new AbortException("reading info file", e);
+        }
+        int endIdx = lines - 100;
+        if(lines > 100) {
+            try(BufferedWriter bw = new BufferedWriter(new FileWriter(infoFile))) {
```

Comment:
It would be safer to make a backup copy of the current info file before overwriting it. As it stands, if an exception (or power outage) occurs while rewriting the info file, you'll lose data.